rules/Php85/Rector/FuncCall/RemoveFinfoBufferContextArgRector.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Php85\Rector\FuncCall;
+
+use PhpParser\Node;
+use PhpParser\Node\Arg;
+use PhpParser\Node\Expr;
+use PhpParser\Node\Expr\CallLike;
+use PhpParser\Node\Expr\FuncCall;
+use PhpParser\Node\Expr\MethodCall;
+use PhpParser\Node\Identifier;
+use PHPStan\Type\ObjectType;
+use Rector\NodeAnalyzer\ArgsAnalyzer;
+use Rector\Rector\AbstractRector;
+use Rector\ValueObject\PhpVersionFeature;
+use Rector\VersionBonding\Contract\MinPhpVersionInterface;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see https://wiki.php.net/rfc/deprecations_php_8_5#deprecate_the_context_parameter_for_finfo_buffer
+ * @see \Rector\Tests\Php85\Rector\FuncCall\RemoveFinfoBufferContextArgRector\RemoveFinfoBufferContextArgRectorTest
+ */
+final class RemoveFinfoBufferContextArgRector extends AbstractRector implements MinPhpVersionInterface
+{
+    public function __construct(
+        private readonly ArgsAnalyzer $argsAnalyzer,
+    ) {
+    }
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition('Remove argument by position by function name', [
+            new CodeSample(
+                <<<'CODE_SAMPLE'
+finfo_buffer($finfo, $fileContents, FILEINFO_NONE, []);
+CODE_SAMPLE
+                ,
+                <<<'CODE_SAMPLE'
+finfo_buffer($finfo, $fileContents, FILEINFO_NONE);
+CODE_SAMPLE
+                ,
+            ),
+        ]);
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [FuncCall::class, MethodCall::class];
+    }
+
+    /**
+     * @param FuncCall $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if ($node->name instanceof Expr) {
+            return null;
+        }
+
+        if ($node instanceof FuncCall && ! $this->isName($node->name, 'finfo_buffer')) {
+            return null;
+        }
+
+        $objectType = new ObjectType(\finfo::class);
```

Comment:
use normal string to avoid prefixed on downgrade:

```suggestion
        $objectType = new ObjectType('finfo');
```

Reply:
Done